state-network.md

```diff
@@ -255,53 +269,297 @@ sedes      := Container(enrs: List[byte_list, max_length=32], payload: byte_list
+
+
+## Gossip
+
+The state network will use a multi stage mechanism to store new and updated state data.
+
+### Overview
+
+The state data is stored in the network in two formats.
+
+- A: Trie nodes
+    - Individual leaf and intermediate trie nodes.
+    - Gossiped with a proof against a recent state root
+    - Stored without a proof not anchored to a specific state root
+- B: Leaf proofs
+    - Contiguous sections of leaf data
+    - Gossiped and stored with a proof against a specific state root
+    - Proof is continually updated to latest state root
+
+The state data first enters the network as trie nodes which are then used to populate and update the leaf proofs.
+
+### Terminology
+
+We define the following terms when referring to state data.  
+
+> The diagrams below use a binary trie for visual simplicity. The same
+> definitions naturally extend to the hexary patricia trie.
+
+
+```
+0:                           X
+                            / \
+                          /     \
+                        /         \
+                      /             \
+                    /                 \
+                  /                     \
+                /                         \
+1:             0                           1
+             /   \                       /   \
+           /       \                   /       \
+         /           \               /           \
+2:      0             1             0             1
+       / \           / \           / \           / \
+      /   \         /   \         /   \         /   \
+3:   0     1       0     1       0     1       0     1
+    / \   / \     / \   / \     / \   / \     / \   / \
+4: 0   1 0   1   0   1 0   1   0   1 0   1   0   1 0   1
+```
+
+
+#### *"state root"*
+    
+The node labeled `X` in the diagram.
+
+#### *"trie node"*
+
+Any of the individual nodes in the trie.
+
+#### *"intermediate node"*
+
+Any of the nodes in the trie which are computed from other nodes in the trie.  The nodes in the diagram at levels 0, 1, 2, and 3 are all intermediate.
+
+#### *"leaf node"*
+
+Any node in the trie that represents a value stored in the trie.  The nodes in the diagram at level 4 are leaf nodes.
+
+#### *"leaf proof"*
+
+The merkle proof which contains a leaf node and the intermediate trie nodes necessary to compute the state root of the trie.
+
+
+
+### Stages
+
+The gossip mechanism is divided up into individual stages which are designed to
+ensure that each individual piece of data is properly disceminated to the DHT
+nodes responsible for storing it, as well as spreading the responsibility as
+evenly as possible across the DHT.
+
+The stages are:
+
+- Stage 1:
+    - Bridge node generates a proof of all new and updated state data from the most recent block and initiates gossip of the individual trie nodes.
+- Stage 2:
+    - DHT nodes receiving trie nodes perform *neighborhood gossip* to spread the data to nearby interested DHT nodes.
```

Comment:
this is a small thing but it might be useful to add a "(defined below)" after neighborhood gossip